sev_serial/src/lib.rs

```diff
@@ -55,68 +61,116 @@ const DATA_TERMINAL_READY_AND_REQUEST_TO_SEND: u8 = 3;
 /// See <https://wiki.osdev.org/Serial_Ports#Line_status_register>.
 const OUTPUT_EMPTY: u8 = 1 << 5;
 
+/// A serial port implementation that uses direct port-based IO.
+pub type RawSerialPort<'a> = SerialPort<'a, PortReadOnly<u8>, PortWriteOnly<u8>, RawIoPortFactory>;
```

Comment:
Why not just use `Port` instead of the ReadOnly and WriteOnly traits?

It looks like neither the RawSerialPort nor the StaticGhcbSerialPort do anything special that would require them to be separate.

Reply:
Not a very good reason, mostly because I did not implement the `PortReader` and `PortWriter` traits on `Port`. Also all of our device drivers use each IO port either for reading or writing, never both.